File: examples/customer-address-jpa/src/test/java/io/zenwave360/example/config/RepositoriesInMemoryConfig.java
package io.zenwave360.example.config;

import io.zenwave360.example.core.outbound.jpa.CustomerRepository;
import io.zenwave360.example.infrastructure.jpa.inmemory.CustomerRepositoryInMemory;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Primary;

// @Configuration
public class RepositoriesInMemoryConfig {

    protected final CustomerRepository customerRepository = new CustomerRepositoryInMemory();

    @Bean
    @Primary
    public <T extends CustomerRepository> T customerRepository() {
        return (T) customerRepository;
    }

}

File: examples/kustomer-address-jpa/src/test/kotlin/io/zenwave360/examples/kotlin/config/RepositoriesInMemoryConfig.kt
package io.zenwave360.examples.kotlin.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Primary

import io.zenwave360.examples.kotlin.repository.jpa.*
import io.zenwave360.examples.kotlin.repository.jpa.inmemory.*


//@Configuration
open class RepositoriesInMemoryConfig {


    protected val customerRepository = CustomerRepositoryInMemory()

    @Bean
    @Primary
    fun customerRepository(): CustomerRepositoryInMemory {
        return customerRepository
    }

}
